pkg/scheduler/flavorassigner/flavorassigner.go

```diff
@@ -589,6 +598,9 @@ func fitsResourceQuota(fName kueue.ResourceFlavorReference, rName corev1.Resourc
 		}
 	}
 	status.append(msg)
+	if borrow := val - rQuota.Nominal; borrow > 0 && mode == Preempt {
```

Comment:
What does mean this condition?

Reply:
Good question. I added it to return `borrow>0` when returning the `Preempt` mode, so that the `FlavorFungability` rules can react to it. 

In particular, when borrowing & preemption is required, and `flavorFungability.whenCanBorrow=TryNextFlavor` and `flavorFungability.whenCanPreempt=Preempt`, then I think we should do `TryNextFlavor`, because preempting on its own would not be enough. If I returned `0` (and not done other modifications in this file), then we would do "Preempt" in this case. WDYT @alculquicondor ?

Your comment made me also think:
1. can the check be simplified? we probably don't need to check `mode == Preempt`
2. I'm missing an analogous check here: https://github.com/kubernetes-sigs/kueue/blob/5fced9e6f0b1323153d2df65816e70a087c8f6f7/pkg/scheduler/flavorassigner/flavorassigner.go#L563
I'm going to experiment with (1.) and (2.).

Reply:
There is another reason we need to return `borrow>0`. This is used for entry ordering in scheduler: https://github.com/kubernetes-sigs/kueue/blob/25630d072e3e58f40414ded524bd9cfdee08c933/pkg/scheduler/scheduler.go#L505

I have modified the implementation. PTAL. Hopefully it is simpler, but I'm open to other suggestions.

Also, the only relevant bit of information is if `borrows()` is true / false. So, I opened another PR to further simplify the flavorassigner in this aspect: https://github.com/kubernetes-sigs/kueue/pull/1498.